server/ServerFSM.cpp

```diff
@@ -1322,7 +1324,15 @@ sc::result MPLobby::react(const LobbyUpdate& msg) {
             m_lobby_data->m_monster_freq   = incoming_lobby_data.m_monster_freq;
             m_lobby_data->m_native_freq    = incoming_lobby_data.m_native_freq;
             m_lobby_data->m_ai_aggr        = incoming_lobby_data.m_ai_aggr;
-            m_lobby_data->m_game_rules     = incoming_lobby_data.m_game_rules;
+
+            // copy only allowed rules
```

Comment:
elaborate comment a bit...

copy rules from incoming lobby data to server lobby data, only if those rules are not locked by the server.

Reply:
Fixed.